### ISSUE TITLE
The edition in cargo.toml does not support versions other than 2015 or 2018, hopefully 2021 or 2022

### ISSUE THREAD

**Issue by keienWang** [Concordium/concordium-rust-smart-contracts#92]
![image](https://user-images.githubusercontent.com/42377006/160223046-d690d028-eefb-4eb5-bb6d-714cc3747758.png)

**Comment by abizjak** [comment]
Which rust compiler version are you using?

**Comment by keienWang** [comment]
> 

rustup 1.24.3

**Comment by keienWang** [comment]
Do you need to update to the latest version?

**Comment by abizjak** [comment]
I don't mean the rustup version, what is your compiler version. If you run `cargo --version`, what does it say?

**Comment by keienWang** [comment]
cargo 1.58.0

**Comment by abizjak** [comment]
Ok. `cargo-concordium` parses the manifest file and the version that is released does not support 2021.

For now the best thing is if you use edition 2018 for your smart contracts.

We will upgrade when time permits. Thank you for the report!

**Comment by abizjak** [comment]
For reference, the issue stems from us using `cargo_toml` dependency `0.8` which does not support `2021` edition. We need to upgrade that dependency when building cargo-concordium.

**Comment by keienWang** [comment]
There is no impact for the time being, just restore the automatic edition to the old version, hope the concordium project will get better and better.

**Comment by abizjak** [comment]
@rimbi Can you please look at this. I believe the fix should be just as I wrote above

> For reference, the issue stems from us using `cargo_toml` dependency `0.8` which does not support `2021` edition. We need to upgrade that dependency when building cargo-concordium.

but we need to make sure it is indeed so.

**Comment by rimbi** [comment]
You are right that  `cago_toml` version must be upgraded. But also, the user must be using `rustc` version >= `1.56`.

**Comment by abizjak** [comment]
`cargo-concordium` uses whatever rust is the default. So they may use `1.56`.
